tiledb/api/src/query/mod.rs

```diff
@@ -46,6 +47,20 @@ pub trait Query<'ctx> {
     fn finalize(self) -> TileDBResult<Array<'ctx>>
     where
         Self: Sized;
+
+    fn subarray(&'ctx self) -> TileDBResult<Subarray<'ctx>> {
```

Comment:
I don't love the overload of the `'ctx` lifetime name, I'd suggest `'query`

Reply:
Actually, it turns out to be not fine. I was pretty sure that `'ctx` was more correct here when I started, but who cares about naming. But now I've had rustc tell me "No" enough times that I'm pretty sure I understand specifically why (except one particular detail I need to go find an answer for).

Regardless, I believe the `'ctx` is correct here because its referring to the Schema reference in the Subarray struct which flows from the same `'ctx` lifetime that parameterizes the Query instance. What you might be thinking of is the lifetime of the Subarray struct instance itself, which is the elided lifetime of the Query instance. This can be confirmed you insert a `drop(query)` after getting the Subarray and then attempting to use the subarray.

The *only* thing I couldn't figure out is why a restricted lifetime wouldn't work. I couldn't decide if this was because the compiler tracks enough to realize that the lifetime parameter for the Query must match the lifetime parameter of the Schema (regardless of what they're called) or if there's a weird issue with restricted lifetimes on generic trait methods. But all I could get in the attempt was for the rustc to tell me that `'ctx: 'query` is shadowing an already defined lifetime `'ctx`. I'm still reading on that one.